### DISCUSSION TITLE
Regarding summarisation for title generation

### DISCUSSION THREAD

**Discussion by QuietRocket** [anse-app/anse#3] [General]
First off, I'd like to say great job on the featv2 revamp! And I like how it's now its own separate repository. Great work 👏 

While reading the code, I noticed there's a functionality to update the conversation title based on the prompt.
I find this feature to be pretty nifty, but wouldn't it be good to make it optional? I just glazed over the repo, but it seems to me like if I use GPT-4, the summary will be made via GPT-4. This consists of its own request copying the prompt and possibly the entire conversation history. It seems like it will chew through API credits pretty quickly.

If I understood this situation correctly, and this indeed could be an issue, there could be two possible fixes:
1. Make a toggle for whether this system should be put in place.
2. For each request, have GPT-4 generate a structured or segmented output containing a small summary / title in the end. That way, only a single request is used for the entire process.

Looking forward to your response,

QuietRocket

**Comment by ddiu8081** [comment]
Thank you for your message!

There are several points to note about title-summarisation:

1. It will always use `gpt-3.5-turbo`, no matter what model you have selected. It uses a separate set of parameters that are not affected by the parameters in the settings panel.  
   https://github.com/anse-app/anse/blob/4dc4029238714ad3171fd2c51217775492d78f24/src/providers/openai/handler.ts#L34-L53
2. The content used to generate the summary includes: system info (not yet implemented), and the first prompt sent by user.
3. It is generated in continuous conversations where the title is not set manually, and is only operated once in the first round.

Making the feature optional is great and I will put it into [roadmap](https://github.com/orgs/anse-app/projects/2).
